File: 2022/Day_1/CPP/cal_counter.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main()
{
    int cal{0};
    int cals={0};
    int max_cals={0};
    string line;
    ifstream InputFile("cals.txt");
    while (getline(InputFile, line))
    {
        if (sscanf(line.c_str(), "%d", &cal) == 1)
        {
            cals+=cal;
        }
        if (line.length() == 0)
        {
            max_cals=(max_cals < cals) ? cals: max_cals;
            cals=0;
        }
    }
    InputFile.close();
    cout<<max_cals<<endl;
}

File: 2022/Day_1/Python/cal_counter.py
cals=0
max_cal=0
with open("cals.txt") as cal_file:
    for line in cal_file:
        if line.strip().isdigit():
            cals+=int(line)
            max_cal=cals if cals>max_cal else max_cal
        else:
            cals=0
print(max_cal)
